eisbuk-admin/src/components/atoms/AttendanceSheet/AttendanceSheetSlot.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import TableBody from "@material-ui/core/TableBody";
+import TableRow from "@material-ui/core/TableRow";
+import Table from "@material-ui/core/Table";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import {
+  CustomerWithAttendance,
+  SlotInterface,
+  SlotInterval,
+} from "@/types/temp";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+
+/** @TODO remove this and import same type when it's merged*/
+export interface Props extends SlotInterface {
+  /**
+   * Record of customers who have booked (or are manually as attended),
+   * keyed by `customerId` and having all of the data for the customer plus
+   * values for `bookedInterval` and `attendedInterval`.
+   */
+  customers: CustomerWithAttendance[];
+}
+
+/**
+ * function that calculates start and end time for slot
+ * @param intervals - slot intervals array
+ * @returns overallInterval - { firstStartTime, lastEndTime}
+ */
+const getTimeString = (intervals: SlotInterval[]): string => {
+  return `${intervals[0].startTime} - ${
+    intervals[intervals.length - 1].endTime
+  }`;
+};
```

Comment:
This isn't right because you're depending on the intervals being sorted, whats more, even if the intervals are sorted, this might happen: `const sortedIntervals = ["09:00-10:00", "09:00-15:00", "10:00-11:00"]` and your result would be `"09:00 - 11:00"` when in fact end time should be `"15:00"`. 
The other thing...why not just take in intervals as they are (record insted of an array) and to the processing internally.